File: backend/models/Society.js
const mongoose = require('mongoose');
const crypto = require('crypto');

function generateCode() {
  return crypto.randomBytes(4).toString('hex').slice(0, 6).toUpperCase();
}

const societySchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    address: { type: String, required: true, trim: true },
    pincode: { type: String, required: true, trim: true },
    area: { type: String, required: true, trim: true },
    city: { type: String, required: true, trim: true },
    state: { type: String, required: true, trim: true },
    code: { type: String, required: true, unique: true, uppercase: true, trim: true, default: generateCode },
    createdBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    admins: { type: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }], default: [] },
  },
  { timestamps: true }
);

const Society = mongoose.model('Society', societySchema);
Society.generateCode = generateCode;

module.exports = Society;

File: lib/models/society.dart
class Society {
  final String id;
  final String name;
  final String address;
  final String pincode;
  final String area;
  final String city;
  final String state;
  final String code;

  const Society({
    required this.id,
    required this.name,
    required this.address,
    required this.pincode,
    required this.area,
    required this.city,
    required this.state,
    required this.code,
  });

  factory Society.fromJson(Map<String, dynamic> json) => Society(
        id: json['_id'] as String,
        name: json['name'] as String,
        address: json['address'] as String,
        pincode: json['pincode'] as String? ?? '',
        area: json['area'] as String? ?? '',
        city: json['city'] as String? ?? '',
        state: json['state'] as String? ?? '',
        code: json['code'] as String,
      );
}
